### PR TITLE
Add constants for MicrosoftAuthService.hello() implementation

### PR DESCRIPTION
**Pull request by heidijinxujia**

### CODE REVIEW

**Comment by heidijinxujia** [comment]
Synced with team on the requirements of the MicrosoftAuthService.connect(). Need to address the change before ready for code review.

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java [hunk 1]
```diff
@@ -609,6 +609,26 @@ private AuthenticationConstants() {
          */
         public static final String BROKER_VERSION = "broker.version";
 
+        /**
+         * The broker protocol version this sdk supports.
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java]: This should be the _highest_ supported version, right?

File: common/src/main/java/com/microsoft/identity/common/internal/request/AcquireTokenSilentOperationParameters.java [hunk 2]
```diff
@@ -32,6 +32,16 @@
     @Expose()
     private boolean mForceRefresh;
 
+    private String mRequiredBrokerProtocolVersion;
+
+    public String getRequiredBrokerProtocolVersion() {
+        return mRequiredBrokerProtocolVersion;
+    }
+
+    public void setRequiredBrokerProtocalVersion(final String requiredBrokerProtocolVersion) {
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/request/AcquireTokenSilentOperationParameters.java]: Nit: `setRequiredBrokerProtocolVersion`

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java [hunk 3]
```diff
@@ -609,6 +609,26 @@ private AuthenticationConstants() {
...
+         */
+        public static final String BROKER_PROTOCOL_VERSION_CODE = "2.0";
+
+        /**
+         * The key of broker protocol version.
+         */
+        public static final String BROKER_PROTOCOL_VERSION_KEY = "broker.protocol.version.name";
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java]: Set in MSAL app config - client is specifying which version they want to use.

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java [hunk 4]
```diff
@@ -609,6 +609,26 @@ private AuthenticationConstants() {
...
+
+        /**
+         * The key of minimum broker protocol version the application requires.
+         */
+        public static final String REQUIRED_BROKER_PROTOCOL_VERSION_KEY = "required.broker.protocol.version.name";
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java]: Set in MSAL app config - client is specifying which version they want to use.

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java [hunk 5]
```diff
@@ -609,6 +609,26 @@ private AuthenticationConstants() {
          */
         public static final String BROKER_VERSION = "broker.version";
 
+        /**
+         * The broker protocol version this sdk supports.
+         */
+        public static final String BROKER_PROTOCOL_VERSION_CODE = "2.0";
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java]: The version of `common`

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java (hunk 3 above)
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java]: The version of the broker protocol understood by MSAL

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java [hunk 6]
```diff
@@ -609,6 +609,26 @@ private AuthenticationConstants() {
...
+
+        /**
+         * The key of minimum broker protocol version the application requires.
+         */
+        public static final String REQUIRED_BROKER_PROTOCOL_VERSION_KEY = "required.broker.protocol.version.name";
+
+        /**
+         * The key of broker protocol version in common.
+         */
+        public static final String COMMON_BROKER_PROTOCOL_VERSION_KEY = "common.broker.protocol.version.name";
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java]: The negotiated protocol version

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java]: Let's call this:
`NEGOTIATED_BROKER_PROTOCOL_VERSION_KEY`

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java (hunk 4 above)
Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java]: Let's call this: `CLIENT_CONFIGURED_MINIMUM_BROKER_PROTOCOL_VERSION`

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java (hunk 3 above)
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java]: Let's call this: `CLIENT_ADVERTISED_MAXIMUM_BP_VERSION_KEY`

**Pull request merged**